pennylane/transforms/mitigate.py

```diff
@@ -12,15 +12,346 @@
+                qml.CNOT(wires=(1,2))
+                qml.RY(x[3], wires=0)
+                qml.RY(x[4], wires=1)
+                qml.RY(x[5], wires=2)
+                return qml.expval(H)
+
+            qnode_ideal = qml.QNode(circuit, dev_ideal)
+            qnode_noisy = qml.QNode(circuit, dev_noisy)
+
+        We can then create folded versions of the noisy qnode and execute them for different scaling factors.
+
+        >>> scale_factors = [1., 2., 3.]
+        >>> folded_res = [qml.transforms.fold_global(qnode_noisy, lambda_)(x) for lambda_ in scale_factors]
+
+        We want to later compare the ZNE with the ideal result.
+
+        >>> ideal_res = qnode_ideal(x)
+
+        ZNE is, as the name suggests, an extrapolation in the noise to zero. The underlyding assumption is that the level of noise is proportional to the scaling factor
+        by artificially increasing the circuit depth. We can perform a polynomial fit using ``numpy`` functions. Note that internally in :func:`~.pennylane.transforms.mitigate_with_zne`
+        a differentiable polynomial fit function :func:`~.pennylane.transforms.poly_extrapolate` is used.
+
+        >>> # coefficients are ordered like coeffs[0] * x**2 + coeffs[1] * x + coeffs[0]
+        >>> coeffs = np.polyfit(scale_factors, folded_res, 2)
+        >>> zne_res = coeffs[-1]
+
+        We used a polynomial fit of ``order = 2``. Using ``order = len(scale_factors) -1`` is also referred to as Richardson extrapolation and implemented in :func:`~.pennylane.transforms.Richardson_extrapolate`.
+        We can now visualize our fit to see how close we get to the ideal result with this mitigation technique.
+
+        .. code-block:: python
+
+            x_fit = np.linspace(0, 3, 20)
+            y_fit = np.poly1d(coeffs)(x_fit)
+
+            plt.plot(scale_factors, folded_res, "x--", label="folded")
+            plt.plot(0, ideal_res, "X", label="ideal res")
+            plt.plot(0, zne_res, "X", label="ZNE res", color="tab:red")
+            plt.plot(x_fit, y_fit, label="fit", color="tab:red", alpha=0.5)
+            plt.legend()
+
+        .. figure:: ../../_static/fold_global_zne_by-hand.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+
+    """
+    # The main intention for providing ``fold_global`` was for it to be used in combination with ``mitigate_with_zne``, which also works with mitiq functions.
+    # To preserve the mitiq functionality, ``mitigate_with_zne`` should get a tape transform.
+    # To make ``fold_global`` also user-facing and work with qnodes, this function is batch_transformed instead, and therefore applicable on qnodes.
+    return [fold_global_tape(circuit, scale_factor)], lambda x: x[0]
+
+
+def fold_global_tape(circuit, scale_factor):
+    r"""
+    This is the internal tape transform to be used with :func:`~.pennylane.transforms.mitigate_with_zne`. For the user-facing function see :func:`~.pennylane.transforms.fold_global`
```

Comment:
```suggestion
    This is the internal tape transform to be used with :func:`~.pennylane.transforms.mitigate_with_zne`. For the user-facing function see :func:`~.pennylane.transforms.fold_global`.
```

Reply:
(optional) Since this is an internal function, you could consider making it private